tensorflow/python/keras/mixed_precision/experimental/autocast_variable.py

```diff
@@ -185,46 +185,66 @@ def constraint(self):
     return self._variable.constraint
 
   def assign(self, value, use_locking=None, name=None, read_value=True):
-    return self._variable.assign(value, use_locking, name, read_value)
+    assign_op = self._variable.assign(value, use_locking, name, read_value)
+    if read_value and resource_variable_ops.is_resource_variable(assign_op):
+      return create_autocast_variable(assign_op)
+    return assign_op
 
   def assign_add(self, delta, use_locking=None, name=None, read_value=True):
-    return self._variable.assign_add(delta, use_locking, name, read_value)
+    assign_op = self._variable.assign_add(delta, use_locking, name, read_value)
+    if read_value and resource_variable_ops.is_resource_variable(assign_op):
+      return create_autocast_variable(assign_op)
+    return assign_op
 
   def assign_sub(self, delta, use_locking=None, name=None, read_value=True):
-    return self._variable.assign_sub(delta, use_locking, name, read_value)
+    assign_op = self._variable.assign_sub(delta, use_locking, name, read_value)
+    if read_value and resource_variable_ops.is_resource_variable(assign_op):
+      return create_autocast_variable(assign_op)
+    return assign_op
 
   def scatter_sub(self, sparse_delta, use_locking=False, name=None):
-    return self._variable.scatter_sub(sparse_delta, use_locking, name)
+    var = self._variable.scatter_sub(sparse_delta, use_locking, name)
+    return create_autocast_variable(var)
```

Comment:
I would also have the logic to not wrap here as well. Even though DistributedVariables currently don't support these scatter methods, they may add it in the future.